### ISSUE TITLE
[OpenStack] Fix NPE during FloatingIP allocation

### ISSUE THREAD

**Issue by mrIncompetent** [kubermatic/kubermatic#2293]
Stack trace:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1744cd4]

goroutine 264 [running]:
github.com/kubermatic/machine-controller/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
    /go/src/github.com/kubermatic/machine-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x1ee7840, 0x3a3b340)
    /usr/local/go/src/runtime/panic.go:502 +0x229
github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/openstack.assignFloatingIPToInstance(0xc420dd8820, 0xc420c90570, 0x24, 0xc420d82bf0, 0x7, 0xc420d82bf7, 0x3, 0xc420972ab0, 0x0, 0x0)
    /go/src/github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/openstack/helper.go:390 +0x724
github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/openstack.(*provider).Create(0xc4209f8a20, 0xc420323600, 0xc420e54660, 0xc420fb0000, 0x3760, 0x0, 0x0, 0x0, 0x0)
    /go/src/github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/openstack/provider.go:428 +0x8a8
github.com/kubermatic/machine-controller/pkg/controller/machine.(*Controller).createProviderInstance(0xc4209d6540, 0x25c43e0, 0xc4209f8a20, 0xc420323600, 0xc420fb0000, 0x3760, 0x0, 0x0, 0x0, 0x0)
    /go/src/github.com/kubermatic/machine-controller/pkg/controller/machine/machine.go:337 +0xca
```

**Acceptance criteria**
- [ ] NPE does not happen
- [ ] Potential linting has been applied to the build to avoid such problems (If there's a linter which would fix it)

**Comment by bashofmann** [comment]
@mrIncompetent as far as I can see this is not fixed in 2.8.x yet because the machine-controller isn't updated

**Comment by mrIncompetent** [comment]
@bashofmann correct, this issue was just the tracking issue for the machine-controller.

We're currently waiting until https://github.com/kubermatic/kubermatic/issues/2279 has been resolved to create a new release of the machine-controller - which then can be used inside kubermatic
